docs/src/conventions.md

```diff
@@ -0,0 +1,123 @@
+
+
+## Code for soundness
+
+We are developing Kani to provide assurance that critical Rust components are verifiably free of certain classes of
+security and correctness issues.
+Thus, it is critical that we provide a verification tool that is sound.
+For the class of errors that Kani can verify, we should not produce a “No Error” result if there was in fact an
+error in the code being verified, i.e., it has no
+“False Negatives”.
+
+Because of that, we bias on the side of correctness.
+Any incorrect modeling
+that may trigger an unsound analysis that cannot be fixed in the short term should be mitigated.
+Here are a few ways how we do that.
+
+### Compilation errors
+
+Make sure to add user-friendly errors for constructs that we can't handle.
+For example, Kani cannot handle panic unwind strategy, and it will fail compilation if the crate uses this
+configuration.
+
+### Internal compiler errors
+
+Even though this doesn't provide users the best experience, you are encouraged to add checks in the compiler for any
+assumptions you make during development.
+Those check can be on the form of `assert!()` or `unreachable!()`
+statement.
+Please provide a meaningful message to help user understand why something failed, and try to explain, at least with 
+a comment, why this is the case.
+
+We don't formally use any specific formal representation of [function contract](https://en.wikipedia.org/wiki/Design_by_contract),
+but whenever possible we do instrument the code with assertions that may represent the function pre- and
+post-conditions to ensure we are modeling the user code correctly.
+
+### Verification errors
+
+In cases where Kani fails to model a certain instruction or local construct that doesn't have a global effect,
+we encode this failure as a verification error.
+I.e., we generate an assertion failure instead of the construct we are modeling using
+[`codegen_unimplemented()`](https://github.com/model-checking/kani/blob/f719b565968568335d9be03ef27c5d05bb8fd0b7/kani-compiler/src/codegen_cprover_gotoc/utils/utils.rs#L50),
+which blocks the execution whenever this construct is reached.
+
+This will allow users to verify their crate successfully as long as
+that construct is not reachable in any harness. If a harness has at least one possible execution path that reaches
+such construct, Kani will fail the verification, and it will mark all checks, other than failed checks, with 
+`UNDETERMINED` status.
+
+### Create detailed issues for "TODO" tasks
+
+It is OK to add "TODO" comments as long as they don't compromise user experience or the tool correctness.
+When doing so, please create an issue that captures the task.
+Add details about the task at hand including any impact to the user.
+Finally, add the link to the issue that captures the "TODO" task as part of your comment.
+
+E.g.:
+```rust
+// TODO: This function assumes type cannot be ZST. Check if that's always the case.
+// https://github.com/model-checking/kani/issues/XXXX
```

Comment:
```suggestion
// <https://github.com/model-checking/kani/issues/XXXX>
```
`cargo doc` wants this, right?

Reply:
I don't think this is doc string though. Does it matter?

Reply:
Sorry, I was under the impression that this was enforced for all URLs, but looks like it isn't. So no need to add this.